Tribler/Core/APIImplementation/LaunchManyCore.py

```diff
@@ -719,43 +725,50 @@ def early_shutdown(self):
             else:
                 self._logger.info("lmc: Dispersy failed to shutdown in %.2f seconds", diff)
 
-        if self.metadata_store is not None:
+        if self.metadata_store:
             yield self.metadata_store.close()
-            self.metadata_store = None
+        self.metadata_store = None
 
         if self.tftp_handler:
             yield self.tftp_handler.shutdown()
-            self.tftp_handler = None
+        self.tftp_handler = None
 
-        if self.session.get_megacache():
+        if self.channelcast_db:
             yield self.channelcast_db.close()
+        self.channelcast_db = None
+
+        if self.votecast_db:
             yield self.votecast_db.close()
+        self.votecast_db = None
+
+        if self.mypref_db:
             yield self.mypref_db.close()
+        self.mypref_db = None
+
+        if self.torrent_db:
             yield self.torrent_db.close()
-            yield self.peer_db.close()
+        self.torrent_db = None
 
-            self.channelcast_db = None
-            self.votecast_db = None
-            self.mypref_db = None
-            self.torrent_db = None
-            self.peer_db = None
+        if self.peer_db:
+            yield self.peer_db.close()
+        self.peer_db = None
 
         if self.mainline_dht:
             from Tribler.Core.DecentralizedTracking import mainlineDHT
             yield mainlineDHT.deinit(self.mainline_dht)
-            self.mainline_dht = None
+        self.mainline_dht = None
 
-        if self.torrent_store is not None:
+        if self.torrent_store:
             yield self.torrent_store.close()
-            self.torrent_store = None
+        self.torrent_store = None
 
-        if self.api_manager is not None:
+        if self.api_manager:
             yield self.api_manager.stop()
-            self.api_manager = None
+        self.api_manager = None
 
-        if self.watch_folder is not None:
+        if self.watch_folder:
             yield self.watch_folder.stop()
-            self.watch_folder = None
+        self.watch_folder = None
```

Comment:
These changes only came to light with adding the shutdown endpoint. Apparently nobody ever noticed any issue earlier.